node-graph/wgpu-executor/src/texture_conversion.rs

```diff
@@ -0,0 +1,269 @@
+use crate::WgpuExecutor;
+use graphene_core::Color;
+use graphene_core::Ctx;
+use graphene_core::color::SRGBA8;
+use graphene_core::ops::Convert;
+use graphene_core::raster::Image;
+use graphene_core::raster_types::{CPU, GPU, Raster};
+use graphene_core::table::{Table, TableRow};
+use graphene_core::transform::Footprint;
+use wgpu::util::{DeviceExt, TextureDataOrder};
+use wgpu::{Extent3d, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages};
+
+/// Uploads CPU image data to a GPU texture
+///
+/// Creates a new WGPU texture with RGBA8UnormSrgb format and uploads the provided
+/// image data. The texture is configured for binding, copying, and source operations.
+fn upload_to_texture(device: &std::sync::Arc<wgpu::Device>, queue: &std::sync::Arc<wgpu::Queue>, image: &Raster<CPU>) -> wgpu::Texture {
```

Comment:
I'm a little annoyed that upload and download look so different, but I guess not much we can do. A function is the correct abstraction for upload and a helper struct for download.